rca/editorial/models.py

```diff
@@ -187,10 +195,11 @@ class EditorialListingPage(BasePage):
 
     def get_editor_picks(self):
         related_pages = []
-        pages = self.related_editorial_pages.all()
-        for value in pages.select_related("page"):
-            if value.page and value.page.live:
-                page = value.page.specific
+        pages = self.related_editorial_pages.all().select_related("page")
+        pages = pages.prefetch_related("page__hero_image", "page__listing_image")
+        for value in pages:
+            page = value.page
+            if page and page.live:
```

Comment:
Could the live check move to the original query, either with `.live()` or `filter(live=True)`?